contracts/extensions/StakedExpenditure.sol

```diff
@@ -129,17 +132,44 @@ contract StakedExpenditure is ColonyExtensionMeta {
     colony.obligateStake(msgSender(), _domainId, stakeAmount);
     uint256 expenditureId = colony.makeExpenditure(_permissionDomainId, _childSkillIndex, _domainId);
 
-    stakes[expenditureId] = Stake({ creator: msgSender(), amount: stakeAmount });
+    stakes[expenditureId] = Stake({ creator: msgSender(), stagedPayment: _stagedPayment, amount: stakeAmount });
     colony.transferExpenditure(expenditureId, msgSender());
 
-    emit ExpenditureMadeViaStake(msgSender(), expenditureId, stakeAmount);
+    emit ExpenditureMadeViaStake(msgSender(), expenditureId, stakeAmount, _stagedPayment);
+  }
+
+  /// @notice Release a staged payment slot
+  /// @param _permissionDomainId The domainId in which the extension has the arbitration permission
+  /// @param _childSkillIndex The index that the `_expenditureId` is relative to `_permissionDomainId`,
+  /// @param _expenditureId The id of the expenditure
+  /// @param _slot The slot being released
+  function releaseStagedPayment(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _expenditureId,
+    uint256 _slot
+  )
+    public
+  {
+    require(stakes[_expenditureId].stagedPayment, "staked-expenditure-not-staged-payment");
+    require(stakes[_expenditureId].creator == msgSender(), "staked-expenditure-not-creator");
+    require(
+      colony.getExpenditure(_expenditureId).status == ColonyDataTypes.ExpenditureStatus.Finalized,
+      "staked-expenditure-not-finalized"
+    );
+
+    bool[] memory mask = new bool[](2); mask[0] = false; mask[1] = true;
+    bytes32[] memory keys = new bytes32[](2); keys[0] = bytes32(0); keys[1] = bytes32(uint256(1));
+    colony.setExpenditureState(_permissionDomainId, _childSkillIndex, _expenditureId, 26, mask, keys, bytes32(0));
+
+    emit StagedPaymentReleased(_expenditureId, _slot);
```

Comment:
Again, one for @arrenv : Is there a situation where you wouldn't want to release and pay? In which case, we could/should call 'claim' here.

Reply:
Yes, if a milestone is never delivered, then it should be possible to not have to release and pay that payment.

Reply:
So, should staged payments have claim delays?

At this stage, it is not required. As from what I understand this would also make it more complicated given that staged payments are essentially really high claim delays.

Reply:
I think what Alex is saying is that we should claim the payment (e.g. send funds to the user) in the same transaction in which we release the claim delay. Otherwise we would force the user to come back to claim the payment, which seems unecessary.

Reply:
To do these we would need to pass an array of any tokens being claimed at that stage. I've gone ahead and added that because I assume that will be preferable.